sdks/wasm/tests/browser/http-spec.js

```diff
@@ -0,0 +1,168 @@
+//describe, beforeAll, it, expext - are the Jasmine default methods
+//karmaHTML is the karma-html package object with the access to all its features
+ 
+describe("The WebAssembly Browser Test Suite",function(){
+    
+    const DEFAULT_TIMEOUT = 1000;
+
+    beforeAll(function(done){
+      //load DOM custom matchers from karma-jasmine-dom package
+      jasmine.addMatchers(DOMCustomMatchers);
+      
+      //lets open our 'http-spec.html' file in the browser by 'index' tag as you specified in 'karma.conf.js'
+      karmaHTML.httpspec.open();
+      
+      //karmaHTML.httpspec.onstatechange fires when the Document is loaded
+      //now the tests can be executed on the DOM
+      karmaHTML.httpspec.onstatechange = function(ready){
+        //if the #Document is ready, fire tests
+        //the done() callback is the jasmine native async-support function
+        if(ready) {
+          karmaHTML.httpspec.document.onRuntimeDone = function ()
+          {
+            done();
+          }
+
+        }
+      };
+
+    });
+    
+    it("should be a real Document object",function(){
+      var _document = karmaHTML.httpspec.document;
+      expect(_document.constructor.name).toEqual('HTMLDocument');
+    });
+
+    it('should support streaming', () => {
+      //karmaHTML.httpspec.document gives the access to the Document object of 'http-spec.html' file
+      var _document = karmaHTML.httpspec.document;
+
+      assert.equal(_document.Module.BINDING.call_static_method("[HttpTestSuite]TestSuite.Program:IsStreamingSupported", []), true);
+    });    
+
+
+    it('should have streaming enabled', () => {
+      //karmaHTML.httpspec.document gives the access to the Document object of 'http-spec.html' file
+      var _document = karmaHTML.httpspec.document;
+
+      assert.equal(_document.Module.BINDING.call_static_method("[HttpTestSuite]TestSuite.Program:IsStreamingEnabled", []), true);
+    });    
+
+    it('should have base path', () => {
+      //karmaHTML.httpspec.document gives the access to the Document object of 'http-spec.html' file
+      var _document = karmaHTML.httpspec.document;
+
+      assert.equal(_document.Module.BINDING.call_static_method("[HttpTestSuite]TestSuite.Program:BasePath", []), "http://localhost:9876/");
+    });    
+
+    it('RequestStream: should return size of Stream with streaming', (done) => {
+      //karmaHTML.httpspec.document gives the access to the Document object of 'http-spec.html' file
+      var _document = karmaHTML.httpspec.document;
+      
+      _document.Module.BINDING.call_static_method("[HttpTestSuite]TestSuite.Program:RequestStream", [true]).then(
+        (result) => 
+        {
+            //console.log("we are here: " + result);
+            assert.equal(result, 500000);
```

Comment:
If we do something like this:
```suggestion
            var error = null;
            try {
              assert.equal(result, 500000);
            } catch (e) {
              error = e;
            }
            done (e);
```
I think the assertions will be properly surfaced to the test results

Reply:
Will do